File: Arcade/Intro/2 - Edge of the Ocean/8 - Matrix Elements Sum/matrixElementsSum.java
int matrixElementsSum(int[][] matrix) {

    //an array of booleans to know when a zero is found
    boolean[] zeroColumnFound = new boolean[matrix[0].length];
    int result = 0;
    
    for(int row = 0; row < matrix.length; row++)
    {
        for(int j = 0; j < matrix[0].length; j++)
        {
            boolean canAcessThisTile = zeroColumnFound[j] == false;
            boolean tileHasValueZero = matrix[row][j] == 0;
            
            if(canAcessThisTile && tileHasValueZero) zeroColumnFound[j] = true;
            else if(canAcessThisTile) result += matrix[row][j];
        }
    }
    
    return result;
}

File: Arcade/Intro/2 - Edge of the Ocean/8 - Matrix Elements Sum/matrixElementsSum.cpp
int matrixElementsSum(std::vector<std::vector<int>> matrix) {
    
    //a vector of booleans to flag when an zero is found in a column
    std::vector<bool> zeroColumnFound (matrix[0].size(), false);
    int result = 0;
    
    for(int i = 0; i < matrix.size(); i++)
    {
        for(int j = 0; j < matrix[0].size(); j++)
        {
            bool canAcessThisTile = zeroColumnFound.at(j) == false;
            bool tileHasValueZero = matrix[i][j] == 0;
            
            if(canAcessThisTile && tileHasValueZero) zeroColumnFound.at(j) = true;
            else if(canAcessThisTile) result += matrix[i][j];
        }
    }
    
    return result;
}
